message_ix/model/MESSAGE/model_core.gms

```diff
@@ -1981,6 +1992,146 @@ RELATION_CONSTRAINT_LO(relation,node,year)$( is_relation_lower(relation,node,yea
+* storage level. Thus, :math:`t^{C}` is a charging technology and :math:`t^{D}` is the corresponding discharger.
+*   .. math::
+*      STORAGE\_CHG_{n,t^S,l,y,h} =
+*          \sum_{\substack{n^L,m,c,h^A \\ y^V \leq y, (n,t^C,t^S,l,y) \sim S^{storage}}} output_{n^L,t^C,y^V,y,m,n,c,l,h^A,h}
+*             \cdot & ACT_{n^L,t^C,y^V,y,m,h^A} \\
+*          - \sum_{\substack{n^L,m,c,h^A \\ y^V \leq y, (n,t^D,t^S,l,y) \sim S^{storage}}} input_{n^L,t^D,y^V,y,m,n,c,l,h^A,h}
+*             \cdot & ACT_{n^L,t^D,y^V,y,m,h^A} \\
+***
+STORAGE_CHANGE(node,storage_tec,level,year,time)$( SUM( (mode,tec,commodity),
+    map_tec_storage_level(node,tec,storage_tec,level,year,time) AND
+    (map_tec_charge(node,tec,mode,commodity,level,year,time) OR
+    map_tec_discharge(node,tec,mode,commodity,level,year,time) ) ) )..
+* change in the content of storage in the examined timestep
+    STORAGE_CHG(node,storage_tec,level,year,time) =E=
+* increase in the content of storage due to the activity of charging technologies
+        SUM( (location,vintage,mode,tec,commodity,time2)$(
+        map_tec_lifetime(node,tec,vintage,year) AND charge_tec(tec)
+        AND map_tec_storage_level(node,tec,storage_tec,level,year,time) ),
+            output(location,tec,vintage,year,mode,node,commodity,level,time2,time)
+            * duration_time_rel(time,time2) * ACT(location,tec,vintage,year,mode,time) )
+* decrease in the content of storage due to the activity of discharging technologies
+        - SUM( (location,vintage,mode,tec,commodity,time2)$(
+        map_tec_lifetime(node,tec,vintage,year) AND discharge_tec(tec)
+        AND map_tec_storage_level(node,tec,storage_tec,level,year,time) ),
+            input(location,tec,vintage,year,mode,node,commodity,level,time2,time)
+            * duration_time_rel(time,time2) * ACT(location,tec,vintage,year,mode,time) );
+
+***
+* Equation STORAGE_BALANCE
+* """""""""""""""""""""""""""""""
+*
+* This equation ensures the commodity balance of storage technologies,
+* where the commodity is shifted between sub-annual time steps within a model period.
+* Here, :math:`h^{A}` is the time step prior to :math:`h`.
+*   .. math::
+*      STORAGE_{n,t^S,l,y,h} \ = init\_storage_{n,t^S,l,y,h} + STORAGE\_CHG_{n,t^S,l,y,h} + \\
+*      STORAGE_{n,t^S,l,y,h^A} \cdot & (1 - storage\_loss_{n,t^S,l,y,h^A}) \\
+***
+STORAGE_BALANCE(node,storage_tec,level,year,time2)$ (
+    SUM(tec, map_tec_storage_level(node,tec,storage_tec,level,year,time2) )
+    AND NOT (time_seq(time2) = 1 ) )..
+* Showing the content of storage at each timestep
+    STORAGE(node,storage_tec,level,year,time2) =E=
+* initial content of storage and change in the content of storage in the examined timestep
+    init_storage(node,storage_tec,level,year,time2)
+    + STORAGE_CHG(node,storage_tec,level,year,time2)
+* storage content in the previous subannual timestep
+    + SUM((time,year2)$map_time_period(year,year2,time,time2),
```

Comment:
I believe that this line is wrong - why would you include in the sum the `STORAGE` variable from the current year but include `year2` in the summation? Also, `year2` is not mentioned in the documentation...

Reply:
@danielhuppmann, sorry for being late in coming back to this. The second index of year (`year2`) was initially intended for cases of shiftIing a commodity between two periods. Later, we decided using storage only for inter temporal commodity shift in one period. So, thanks for noticing this, `year2` was extra and I removed it.